packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -695,14 +695,24 @@ export class Instance extends Resource implements IInstance {
   }
 
   /**
-   * Apply CloudFormation update policies for the instance
+   * Apply CloudFormation update policies for the instance.
+   *
+   * If both initOptions.timeout and resourceSignalTimeout are specified,
+   * timeout is summed together, else timeout is set to the value of the specified field.
    */
   private applyUpdatePolicies(props: InstanceProps) {
+    if (props.initOptions?.timeout && props.resourceSignalTimeout) {
+      Annotations.of(this).addWarningV2('@aws-cdk/aws-ec2:setSetimeout', 'Both initOptions.timeout and resourceSignalTimeout fields are set, timeout is summed together. It is suggested that only one of the two fields is set');
+    }
+
     if (props.resourceSignalTimeout !== undefined) {
       this.instance.cfnOptions.creationPolicy = {
         ...this.instance.cfnOptions.creationPolicy,
         resourceSignal: {
-          timeout: props.resourceSignalTimeout && props.resourceSignalTimeout.toIsoString(),
+          ...this.instance.cfnOptions.creationPolicy?.resourceSignal,
+          timeout: props.initOptions?.timeout ?
+            props.resourceSignalTimeout.plus(props.initOptions?.timeout).toIsoString() :
```

Comment:
can you double check this for me: i don't think you need the `?` on this line because typescript should be smart enough to know `props.initOptions` must exist to be on this side of the ternary operator.

Reply:
Yes, you are right! I have updated this.